sycl/test-e2e/Graph/Threading/finalize.cpp

```diff
@@ -12,9 +10,34 @@
 // ZE_DEBUG=4 testing capability.
 
 #include "../graph_common.hpp"
+#include <detail/graph_impl.hpp>
```

Comment:
Why did we keep this test as a `test-e2e` test but move the other tests which use `graph_impl` to unittests? Is there a reason this can't be a unittest?

Reply:
This tests submits the finalized graph to a queue. So we need a device for this test. 
And I prefer to not remove graph submission to test the thread safety property of "ext_oneapi_graph()".

Reply:
Makes sense 👍